Jenkinsfile

```diff
@@ -283,6 +351,76 @@ pipeline {
                         }
                     }
                 }
+                stage('experimental-master') {
```

Comment:
Opened https://github.com/moby/moby/pull/39711 to try this approach

Reply:
Thanks for trying out an alternative approach, but the reason to have a separate stage with its own agent for running on master would be to keep the overall time under 2 hrs. Granted, that means we will have a job run on the master branch use two z machines instead of just one z machine for a PR job, but it will keep the master job running under 1 hour in total.

If we run the integration and integration-cli tests on the same node for z machines, it will sometimes take over 2 hrs so we will then have to change the timeout settings to be different for a PR and for a master job.

The different filenames for `z-bundles.tar.gz` and `z-master-bundles.tar.gz` is to help a user distinguish the difference between the two if they download both from the same artifacts page of the jenkins job.

I agree there is a bit of duplicate code going on in the `Jenkinsfile` because of this, but I think we could simplifying into Makefile targets which is what a human would want anyways to easily reproduce a failure observed in a jenkins job.

For now, concentrating on getting jenkins jobs below 1 hour as top priority. The 1 hour we save now will help our future PRs to make the code cleaner.

Reply:
Ah, gotcha; it's to run them in parallel. 

Perhaps rename them to what they actually do, instead of the branch they're running on? So `XX-integration`, `XX-integration-cli`

We should try to use the same approach as is done on Janky (if possible), but current Power and Z machines have less memory

This https://github.com/moby/moby/pull/39708 (or similar) would also help reduce the duplication in code, and keep the Jenkinsfile a bit more manageable